### PR TITLE
name single threads of kafka live event listener

### PR DESCRIPTION
**Pull request by kishansairam9**

### CODE REVIEW

File: kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ class ListenerThreadFactory implements ThreadFactory {
   private final String name;
 
   public ListenerThreadFactory(String consumerName) {
-    this.name = "kafka-live-event-listener-" + consumerName;
+    this.name = "kafka-live-event-listener-" + consumerName + "-" + System.currentTimeMillis();
```
Comment (kotharironak) [inline kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java:104]: Is it common pattern to add time in name?

Comment (kishansairam9) [reply kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java:104]: I added here intentionally to detect if we have multiple threads for same consumer name

File: kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java [hunk 2]
```diff
@@ -84,10 +85,26 @@ public Builder<K, V> withExecutorService(
...
           cleanupExecutor);
     }
   }
 }
+
+class ListenerThreadFactory implements ThreadFactory {
+  private final String name;
+
+  public ListenerThreadFactory(String consumerName) {
+    this.name = "kafka-live-event-listener-" + consumerName + "-" + System.currentTimeMillis();
```
Comment (kotharironak) [inline kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java:104]:
```suggestion
    this.name = "kafka-live-event-listener-" + consumerName + "-" + String.valueOf(System.currentTimeMillis());
```

File: kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java (hunk 1 above)
Comment (kishansairam9) [reply kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java:104]: Ideally for one consumer there should be only one live event listener hence one thread only. But if it is wrongly used there might be more than one, which we want to detect and fix

File: kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java (hunk 2 above)
Comment (kishansairam9) [reply kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java:104]: image
it is not needed

File: kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java (hunk 1 above)
Comment (kishansairam9) [reply kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java:104]: > If, its more than that, who is caller? How would we fix this? I think, is this called by our application code or by kafka framework itself?

This is used by application code, mainly to listen to change events. If we see more than one thread for a consumer name that means we are rebuilding live event listener under some condition, which is wrong. To fix, you might need to organise the building of live event listener to the place where code is exactly once executed - guice singleton or build topology and pass it down appropriately

**Review by kotharironak — APPROVED** [review]

**Pull request merged**